src/goto-symex/symex_start_thread.cpp

```diff
@@ -20,16 +20,16 @@ void goto_symext::symex_start_thread(statet &state)
 
   // we don't allow spawning threads out of atomic sections
   // this would require amendments to ordering constraints
-  if(state.atomic_section_id!=0)
-    throw "start_thread in atomic section detected";
+  INVARIANT(state.atomic_section_id==0,
+    "start_thread in atomic section detected");
```

Comment:
This could be a PRE_CONDITION, maybe?

Reply:
The message might be better phrased as something like "start_thread not supported in atomic section" ?

Reply:
While I agree with you, I find the message to be useful (in that it contains information that can't be transmitted through the condition checked easily), and making this a `PRECONDITION` would require the message to be killed. I don't believe this to be a good idea, this is why I kept it as an `INVARIANT`.

However, I can change this if there are strong feelings about it.